### ISSUE TITLE
alloc support

### ISSUE THREAD

**Issue by alexbuzzbee** [jamesmunns/postcard#1]
As it is now, Postcard uses `heapless`. While this is fine for many embedded applications, for some applications `alloc` is available and a better option. It would be nice to have a feature `alloc` that would cause postcard to use `alloc` instead of `heapless`.

**Comment by jamesmunns** [comment]
I definitely support this! We do have a `std` feature, which uses the std library, and I would welcome a PR to add alloc support as well!

**Comment by Dentosal** [comment]
I needed a version of this crate with `alloc` support, so I forked it and implemented the support. The new fork is available at [Dentosal/pinecone](https://github.com/Dentosal/pinecone). Unfortunately for this crate, I removed flavors as well. I have no use for them, and removal simplified alloc support implementation. This means I will not create a PR. Still, feel free to borrow code from my version if something there seems useful.
